proxies/nodejs/handlers/client.ts

```diff
@@ -19,11 +20,33 @@ export const handleClient = async (ctx: Koa.ParameterizedContext) => {
         return ctx
     }
     try {
-        dataStore.clients[clientId] = initialize(sdkKey, { ...options,  enableCloudBucketing })
+        let asyncError
+        let client: DVCClient | DVCCloudClient
+        if (!enableCloudBucketing) {
+            client = initialize(sdkKey, { ...options })
+            if (waitForInitialization) {
+                try {
+                    await client.onClientInitialized()
+                } catch (e) {
+                    asyncError = e
+                }
+            }
+        } else {
+            client = initialize(sdkKey, { ...options, enableCloudBucketing: true })
+        }
+        dataStore.clients[clientId] = client
         ctx.status = 201
         ctx.set('Location', `client/${clientId}`)
-        ctx.body = {
-            message: 'success'
+
+        if (asyncError) {
+            ctx.status = 201
```

Comment:
the status is already set above so I don't think you need this here. Unless this should be a 200